File: src/main/java/com/zunza/buythedip/crypto/entity/Crypto.java
package com.zunza.buythedip.crypto.entity;

import com.zunza.buythedip.common.BaseEntity;

import jakarta.persistence.CascadeType;
import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.FetchType;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.OneToOne;
import lombok.AccessLevel;
import lombok.Getter;
import lombok.NoArgsConstructor;

@Getter
@Entity
@NoArgsConstructor(access = AccessLevel.PROTECTED)
public class Crypto extends BaseEntity {
	@Id
	@GeneratedValue(strategy = GenerationType.IDENTITY)
	private Long id;

	@Column(nullable = false, length = 50)
	private String name;

	@Column(nullable = false, length = 20)
	private String symbol;

	@Column
	private String logo;

	@Column(nullable = false)
	private Double tickSize;

	@OneToOne(
		mappedBy = "crypto",
		cascade = CascadeType.ALL,
		orphanRemoval = true,
		fetch = FetchType.LAZY
	)
	private CryptoMetadata metadata;
}

File: src/main/kotlin/com/zunza/buythedip/crypto/entity/Crypto.kt
package com.zunza.buythedip.crypto.entity

import com.zunza.buythedip.common.BaseEntity
import jakarta.persistence.CascadeType
import jakarta.persistence.Column
import jakarta.persistence.Entity
import jakarta.persistence.FetchType
import jakarta.persistence.GeneratedValue
import jakarta.persistence.GenerationType
import jakarta.persistence.Id
import jakarta.persistence.JoinColumn
import jakarta.persistence.OneToOne

@Entity
class Crypto(
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    val id: Long = 0,

    @Column(nullable = false, length = 20)
    val name: String,

    @Column(nullable = false, length = 20)
    val symbol: String,

    @Column
    val logo: String? = null,

    @OneToOne(
        mappedBy = "crypto",
        cascade = [CascadeType.ALL],
        orphanRemoval = true,
        fetch = FetchType.LAZY
    )
    val metadata: CryptoMetadata? = null
) : BaseEntity()
